### PR TITLE
❗#2267 Submitted Data Error Banner

### PR DESCRIPTION
**Pull request by demariadaniel**
# Description of changes
Adds Error Banner to Submitted Data Page
Groups errors by # of donors w/ same error on same field
Errors displayed for each Entity tab w/ side bar notification badge
Links to Data Dictionary + Program Submission Page
Default limit to 5 error groups w/ pagination
Download Error Report button

https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/icgc-argo/platform-ui/2267

## Type of Change

- [ ] Bug
- [ ] Styling
- [x] New Feature

## Checklist before requesting review

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: components/pages/submission-system/ErrorNotification.tsx [hunk 1]
```diff
@@ -69,18 +69,20 @@ export default <Error extends { [k: string]: any }>({
   columnConfig,
   onClearClick,
   tsvExcludeCols = [],
+  tableProps,
 }: {
   level: NotificationVariant;
   title: string;
-  subtitle: string;
+  subtitle: string | ReactNode;
```
Comment (ciaranschutte) [inline components/pages/submission-system/ErrorNotification.tsx:76]: ReactNode covers string
`subtitle: ReactNode` is sufficient

File: components/pages/submission-system/program-submitted-data/common.tsx [hunk 2]
```diff
@@ -17,14 +17,30 @@
...
+  followUps = 'followUps',
 }
 
+export const CoreCompletionFields = Object.values(CoreCompletionEntities);
+
+export type CoreCompletion = {
+  CoreCompletionEntities: number;
+};
+
+export type CompletionStats = {
```
Comment (ciaranschutte) [inline components/pages/submission-system/program-submitted-data/common.tsx:36]: I really love seeing all your typings. Thanks for adding these, really makes TS project a lot better.

Comment (demariadaniel) [reply components/pages/submission-system/program-submitted-data/common.tsx:36]: Oh many they really do help the developer experience, the more we can use the better

File: components/pages/submission-system/ErrorNotification.tsx (hunk 1 above)
Comment (demariadaniel) [reply components/pages/submission-system/ErrorNotification.tsx]: Thanks, updated

**Review by ciaranschutte — APPROVED** [review]

**Pull request merged**